### ISSUE TITLE
Agent naming mismatch after endpoints were renamed

### ISSUE THREAD

**Issue by SystemsWest** [wazuh/wazuh#16083]
|4.3.10-1|Wazuh Manager|Manager & Agent|OVA|Windows 10|
|---|---|---|---|---|
| X.Y.Z-rev | Wazuh component | Manager/Agent | Packages/Sources | OS version |

I'm just starting out so bear with me.

I successfully deployed several Windows agents via the provided PowerShell script and they were all visible on the Wazuh dashboard as well as from the manage_agent command-line utility. At this point, this is all the progress I have made.

However, shortly after I finished this, my colleague decided to change the name of three endpoints. At this point, the manage_agent utility and the Wazuh Dashboard still showed the previous name of the endpoints.

I have found the agent name and ID in the .agent-info file on the individual endpoints. After some research and testing, I deleted the agent on these three endpoints and from the manage_agent utility, confirmed that the ossec-agent folder was deleted on the endpoint and then removed the agent from the database. So far, so good. Then I reinstalled the agent (again via PowerShell script) with the new name. The ossec-agent folder was recreated with the correct endpoint name in the .agent-info file. After manually starting the Wazuh service, it came up.

Now, the renamed endpoint name reflects correctly in manage_agent, but not on the Dashboard. Because of the agent deletion, the agent ID also changed.

Is there a simpler way to address endpoint name changes?

**Comment by MiguelazoDS** [comment]
Hi @SystemsWest, 

Let me get this straight. First of all, if I didn't get you wrong, you don't need to reinstall or delete the agent if the hostname was changed. Using the manage_agents tool you could remove those old entries (manage_agents -r <id>). 

You can use `agent_control -l` to list all the agents connected to the manager and check their status as well. 

For example, this may be similar to your case.

![image](https://user-images.githubusercontent.com/13010397/216720267-60884791-ea31-42ec-ac07-bff5eadb44d8.png)

Probably you need to delete those old (disconnected) agents, because they may be interfering to register the new ones, because, in fact, they are the same agents with different names. 

What is the state of the agents in the UI? They should be disconnected as well.

I think that new agents may be trying to re-register but the force options to replace an agent are stopping that. 

You can read more about that here: https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/auth.html#force

Let me know how it goes. Regards!

**Comment by SystemsWest** [comment]
Thank you for responding.

agent_control and manage_agents (executed via CLI on the Wazuh server) both show the correctly named agent and new ID after the endpoint name change.

The .agent-info file on the endpoint has the correct endpoint name.

The Wazuh dashboard still shows the endpoint name BEFORE the name change.

What's curious is that none of them were disconnected... I would have thought that, but they weren't. The endpoints are all DHCP, manage_agents shows the IP as ANY, but they were connected.

Anyway, following the documentation link you provided, I put the <force></force> into the ossec.conf (right before the first </ossec_config> tag) on the server and restarted wazuh-manager... it failed to start. I tried commenting it out... no change. I took it out... the service restarted.

I had already restarted the server, I had restarted the services (dashboard, anyway... I don't recall restarting manager; I've created a script now to restart the services), I had restarted endpoint agents and devices... no change.

Today, I applied CentOS updates and restarted the server... I probably should have tried that sooner. Anyway, now the names are correct in Dashboard and at the CLI. I'm left to try and figure out why...


Ward Barker

___


SystemsWestEngineers.com<http://systemswestengineers.com/>      Main (541) 342-7210

________________________________

**Comment by MiguelazoDS** [comment]
Well, that's strange, the `agent_control` tool and the UI take the information from the same place. Those services may not be working as expected and the restart make them work again. Is this the OVA? 

Althouth seems to not be related to the authd configuration. The `force` option block belongs to the `authd` module, you can not put it in a random place in the ossec.conf file. That's why it fails at start. Those tag are not recognized outside `authd`.  

Anyway, glad you could make it work.

**Comment by SystemsWest** [comment]
I thought the same thing, that the agent name display came from the same place, which didn't help my confusion.

I'm also really glad I could get it work, but you provided some useful information in the process. Regarding use of the <force> section. .. that's my mistake: is that something can safely insert AND leave in the ossec.conf?

I've got a lot to learn here...

Ward Barker

IT Coordinator

___



SystemsWestEngineers.com<http://systemswestengineers.com/>      Main (541) 342-7210<tel:+15413427210>

________________________________

**Comment by MiguelazoDS** [comment]
Hi @SystemsWest, 

Yes, it is meant to configure the agent replacement logic. Although this case may not be necessary and it is not the cause of this issue, I will try to reproduce this to better understand what could be happening. But since it was fixed by a restart, probably something was wrong with the UI services or there's some thread that is not updating the information as it should, which is strange cause it works as expected when you disconnect an agent or add a new one.

**Comment by MiguelazoDS** [comment]
Something like this I was thinking before related to force block. Which is undesired behavior too.

Manager reports the agent as connected, but we can clearly see that the agent couldn't be replaced due to force block configuration, because the agent is active.
![image](https://user-images.githubusercontent.com/13010397/217843620-59fdec22-8ba0-4acb-97bb-51f5c326785f.png)
From the agent side, it is rejected by the manager.
![image](https://user-images.githubusercontent.com/13010397/217843686-9f7a8744-a2e3-4373-8064-d0257400d5fa.png)

I will update with more information for your case

**Comment by MiguelazoDS** [comment]
Yes, that's the way to go, after some testing, the hostname change is not reflected in the manager, and the agent in my case is connected with the previous one. The reason is that the client.keys file is not modified, ergo, the key is still valid, and the agent remains connected. Besides that detail, the manager keeps working as expected, but with the older name. 

The process to change the hostname was through the "Rename this PC" button in the system configuration.  

So to reflect the change, the process is the one you did, so the key now is generated with the new hostname.

**Comment by SystemsWest** [comment]
Okay, thank you for that additional information.

Ward Barker

IT Coordinator

___



SystemsWestEngineers.com<http://systemswestengineers.com/>      Main (541) 342-7210<tel:+15413427210>

________________________________

**Comment by MiguelazoDS** [comment]
Thanks to your sir! I'm closing this issue since the behavior is expected.
